include/RAJA/util/SoAPtr.hpp

```diff
@@ -165,6 +168,77 @@ class SoAPtr<RAJA::reduce::detail::ValueLoc<T, IndexType, doing_min>, mempool, a
   second_type* mem_idx = nullptr;
 };
 
+/*!
+ * @brief Specialization for RAJA::expt::ValLoc.
+ */
+template <typename T, typename IndexType, typename mempool, typename accessor>
+class SoAPtr<RAJA::expt::ValLoc<T, IndexType>, mempool, accessor>
```

Comment:
Do we have to worry about ValOp<ValLoc<...>>?

Reply:
No, by this point, the `SoAPtr` will have been instantiated with a `T`. That `T` will either be int/float/double, or `ValLoc`. This is how we avoid passing in the entire new ValOp object.